File: src/frontend/src/lib/domain/terminal.ts
export type EntryRecord = {
  entryId: string
  filename: string
  capturedAt: string
  inPool: boolean
}

export type ScreenshotRecord = {
  filename: string
  capturedAt: string
  inPool: boolean
}

export type TerminalTab = 'entries' | 'screenshots' | 'logs'

export type CollectDateMode = 'today' | 'fixed'

export type TerminalManagementRecord = {
  id: string
  ipAddress: string
  displayName: string | null
  isOnline: boolean | null  // null = 確認中
  autoCollect: boolean
  collectDateMode: CollectDateMode
  collectDate: string  // 'YYYY-MM-DD'
}

export type PoolStats = {
  fileCount: number
  totalBytes: number
}

export type CleanupMode = 'before' | 'range' | 'all'

export type FileContentType = 'text' | 'table' | 'json'

export type EntryFile = {
  name: string        // 内部ファイル名（API識別用）
  label: string       // 日本語表示名
  type: FileContentType
  sizeBytes?: number
}

File: src/backend/app/schemas/terminal.py
"""端末機能のスキーマ定義"""

from pydantic import BaseModel


class EntryRecord(BaseModel):
    entry_id: str
    filename: str
    captured_at: str  # "YYYY-MM-DD HH:MM:SS"
    in_pool: bool     # プールに採取済みか


class ScreenshotRecord(BaseModel):
    filename: str
    captured_at: str  # "YYYY-MM-DD HH:MM:SS"
    in_pool: bool


class TerminalManagementRecord(BaseModel):
    id: str
    ip_address: str
    display_name: str | None
    is_online: bool | None   # None = 確認中
    auto_collect: bool
    collect_date_mode: str   # "today" | "fixed"
    collect_date: str        # "YYYY-MM-DD"


class PoolStats(BaseModel):
    file_count: int
    total_bytes: int


class CleanupRequest(BaseModel):
    mode: str                      # "before" | "range" | "all"
    date_from: str | None = None   # "YYYY-MM-DD"
    date_to: str | None = None     # "YYYY-MM-DD"


class CleanupResult(BaseModel):
    deleted_count: int
    freed_bytes: int
